### PR TITLE
Testing: Update no gpu test to check otel config

### PR DESCRIPTION
**Pull request by LujieDuan**
## Description
Update `TestNoNvmlOtelReceiverWithoutGpu` to check the generated otel config and assert that nvml receiver is not in the generated config when the VM has no GPU. 

## Related issue
[b/341109955](http://b/341109955)

## How has this been tested?
Integration test passing. 
Tested locally to make sure the test reports correct error when the generated otel config is not presented in the VM. 

## Checklist:
- Unit tests
  - [x] Unit tests do not apply.
  - [ ] Unit tests have been added/modified and passed for this PR.
- Integration tests
  - [ ] Integration tests do not apply.
  - [x] Integration tests have been added/modified and passed for this PR.
- Documentation
  - [x] This PR introduces no user visible changes.
  - [ ] This PR introduces user visible changes and the corresponding documentation change has been made.
- Minor version bump
  - [x] This PR introduces no new features.
  - [ ] This PR introduces new features, and there is a separate PR to bump the [minor version](https://github.com/GoogleCloudPlatform/ops-agent/blob/master/VERSION) since the last [release](https://github.com/GoogleCloudPlatform/ops-agent/releases) already.
  - [ ] This PR bumps the version.

### CODE REVIEW

File: integration_test/ops_agent_test/main_test.go [hunk 1]
```diff
@@ -4732,12 +4739,13 @@ func TestNoNvmlOtelReceiverWithoutGpu(t *testing.T) {
...
-			t.Error("expected no logs to contain Nvidia or nvml when the instance has no gpu")
-		} else if !strings.Contains(err.Error(), "not found, exhausted retries") {
-			t.Fatalf("unexpected error: %v", err)
+		otelConfig, err := retrieveOtelConfig(ctx, logger, vm)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if strings.Contains(otelConfig, "nvml") {
+			t.Errorf("expects the nvml receiver not in the Otel configuration for VMs without GPU. Otel config: " + otelConfig)
```
Comment (franciscovalentecastro) [inline integration_test/ops_agent_test/main_test.go:4748]: [optional][nit] Similar to the previous comment, here `t.Errorf` is not necessary if we are concatenating the strings. I suggest to use only one, formatting with `Errorf` or string concatenation.

File: integration_test/gce/gce_testing.go [hunk 2]
```diff
@@ -870,6 +870,17 @@ func UploadContent(ctx context.Context, logger *log.Logger, vm *VM, content io.R
...
 }
 
+// RetrieveContent retrieves the file content from the the given file path from
+// the remote VM
+func RetrieveContent(ctx context.Context, logger *log.Logger, vm *VM, remotePath string) (content string, err error) {
+	if IsWindows(vm.ImageSpec) {
+		out, err := RunRemotely(ctx, logger, vm, fmt.Sprintf("Get-Content -Path '%s' -Raw", remotePath))
+		return out.Stdout, err
+	}
+	out, err := RunRemotely(ctx, logger, vm, fmt.Sprintf("sudo cat "+remotePath))
```
Comment (franciscovalentecastro) [inline integration_test/gce/gce_testing.go:880]: [optional][nit] Here `fmt.Sprintf` is not necessary if we are concatenating the strings. I suggest to use only one, formatting with `Sprintf` or string concatenation.

**Review by franciscovalentecastro — APPROVED** [review]
Added two optional comments, aside from that LGTM!

File: integration_test/ops_agent_test/main_test.go (hunk 1 above)
Comment (LujieDuan) [reply integration_test/ops_agent_test/main_test.go]: Done!

File: integration_test/gce/gce_testing.go (hunk 2 above)
Comment (LujieDuan) [reply integration_test/gce/gce_testing.go]: Done!

**Pull request merged**